templates/common/_base/files/ofport-request.yaml

```diff
@@ -0,0 +1,114 @@
+    if [ "$PORT_CONNECTION_UUID" == "" ]; then
+        exit 0
+    fi
+    
+    # Get the port's slave-type. If this is not an ovs-bridge, then exit
+    PORT_OVS_SLAVE_TYPE=$(nmcli -t -f connection.slave-type conn show "$PORT_CONNECTION_UUID" | awk -F ':' '{print $NF}')
+    if [ "$PORT_OVS_SLAVE_TYPE" != "ovs-bridge" ]; then
+        exit 0
+    fi
+    
+    # Get the port's master = bridge name
+    BRIDGE_NAME=$(nmcli -t -f connection.master conn show "$PORT_CONNECTION_UUID" | awk -F ':' '{print $NF}')
+    # Limit this to br-ex and br-ex1 only. If one wanted to enable this for all OVS bridges,
+    # the condition would be: if [ "$BRIDGE_NAME" == "" ]; then
+    if [ "$BRIDGE_NAME" != "br-ex" ] && [ "$BRIDGE_NAME" != "br-ex1" ]; then
+        exit 0
+    fi
+    
+    CONFIGURATION_FILE=/run/ofport_requests.${BRIDGE_NAME}
+    
+    # Declare a new map. If CONFIGURATION_FILE exists, source entries from there
+    declare -A INTERFACES
+    if [ -f $CONFIGURATION_FILE ]; then
+        echo "Sourcing configuration file '${CONFIGURATION_FILE}' with contents: $(cat $CONFIGURATION_FILE)"
+        source $CONFIGURATION_FILE
+    fi
+    
+    # get_interface_ofport_request will return
+    # * either: the current ofport assignment for the port if no interface has claimed this ofport number, yet
+    # * or:     the lowest available free ofport number
+    function get_interface_ofport_request() {
+        # Build a map that only contains the currently reserved ofport_requests
+        declare -A ofport_requests
+        for interface_name in "${!INTERFACES[@]}"; do
+    	ofport_requests[${INTERFACES[$interface_name]}]=${INTERFACES[$interface_name]}
+        done
+    
+        # Get the current ofport number assignment
+        local current_ofport=$(ovs-vsctl get Interface "$INTERFACE_NAME" ofport)
+    
+        # If the current ofport number is still free, use it
+        if ! [ "${ofport_requests[$current_ofport]+a}" ]; then
+    	    echo $current_ofport
+    	    return
+        fi
+    
+        # If the current ofport number is not free, return the lowest free entry
+        i=0
+        for i in {1..65000}; do
+            if ! [ "${ofport_requests[$i]+a}" ]; then
+                echo $i
+                return
+            fi
+        done
+    
+        # if we still cannot find an ID, exit with an error
+        echo "Impossible to find an ofport ID for interface $INTERFACE_NAME" >&2
+        exit 1
+    }
+    
+    # If INTERFACES[INTERFACE_NAME] exists, use that value
+    # If INTERFACES[INTERFACE_NAME] does not exists, use get_interface_ofport_request to get a new value
+    if ! [ "${INTERFACES[$INTERFACE_NAME]+a}" ]; then
+    	INTERFACES[$INTERFACE_NAME]=$(get_interface_ofport_request)
+    fi
+    # Set ofport_request according to INTERFACES[INTERFACE_NAME]
+    ovs-vsctl set Interface "$INTERFACE_NAME" ofport_request=${INTERFACES[$INTERFACE_NAME]}
```

Comment:
Are we guaranteed the interface exists at this point?

Reply:
I threw in https://github.com/openshift/machine-config-operator/pull/2941/commits/9c63e55af5cca74291ae11a00d55ec17f31bb0a6#diff-bfb1096d30a23088db321d88004b44a807716f56cad84e920f3e5451f3d6f1e0R65   as a sanity check to exit gracefully if the interface does not exist; so if we get here, we are guaranteed that the interface exists (unless we had a race and some other entity pulled out the interface in the meantime, of course, but I think we can safely assume that only nm is managing the interface)